plugins/destination/kafka/client/client.go

```diff
@@ -25,23 +24,15 @@ type Client struct {
 	spec       specs.Destination
 	pluginSpec Spec
 	metrics    destination.Metrics
-
-	csvTransformer         *csv.Transformer
-	csvReverseTransformer  *csv.ReverseTransformer
-	jsonTransformer        *json.Transformer
-	jsonReverseTransformer *json.ReverseTransformer
+	filetype   *filetypes.Client
```

Comment:
embed

Reply:
Can't embed for kafka, as it uses `Write` rather than `BatchedWrite`

Reply:
you can but will have to impl extra method